jacobian_follower/include/jacobian_follower/jacobian_follower.hpp

```diff
@@ -56,69 +111,72 @@ class JacobianFollower {
+                                moveit_msgs::RobotState const &start_state_msg,
+                                std::vector<PointSequence> const &grippers, double max_velocity_scaling_factor,
+                                double max_acceleration_scaling_factor);
 
-  PlanResult plan(planning_scene_monitor::LockedPlanningSceneRW &planning_scene, std::string const &group_name,
-                  std::vector<std::string> const &tool_names,
-                  std::vector<Eigen::Vector4d> const &preferred_tool_orientations,
-                  robot_state::RobotState const &start_state, std::vector<std::vector<Eigen::Vector3d>> const &grippers,
-                  double max_velocity_scaling_factor, double max_acceleration_scaling_factor);
+  PlanResultMsg plan_from_scene_and_state(std::string const &group_name, std::vector<std::string> const &tool_names,
+                                          std::vector<Eigen::Vector4d> const &preferred_tool_orientations,
+                                          moveit_msgs::RobotState const &start_state_msg,
+                                          moveit_msgs::PlanningScene const &scene_msg,
+                                          std::vector<PointSequence> const &grippers,
+                                          double max_velocity_scaling_factor, double max_acceleration_scaling_factor);
 
   // TODO: Return std::vector<std_msgs/JointState.msg> to avoid ambiguity in the joint ordering?
   std::vector<std::vector<double>> compute_IK_solutions(geometry_msgs::Pose target_pose, const std::string &group_name) const;
 
   // TODO: Accept std_msgs/JointState.msg to avoid ambiguity in the joint ordering?
   geometry_msgs::Pose computeFK(const std::vector<double> &joint_angles, const std::string &group_name) const;
 
-  PlanResult moveInRobotFrame(planning_scene_monitor::LockedPlanningSceneRW &planning_scene,
-                              std::string const &group_name, std::vector<std::string> const &tool_names,
-                              EigenHelpers::VectorQuaterniond const &preferred_tool_orientations,
-                              robot_state::RobotState const &start_state, PointSequence const &target_tool_positions);
+  PlanResult plan(JacobianTrajectoryCommand traj_command);
 
-  PlanResult moveInWorldFrame(planning_scene_monitor::LockedPlanningSceneRW &planning_scene,
-                              std::string const &group_name, std::vector<std::string> const &tool_names,
-                              EigenHelpers::VectorQuaterniond const &preferred_tool_orientations,
-                              robot_state::RobotState const &start_state, PointSequence const &target_tool_positions);
+  PlanResult moveInWorldFrame(JacobianWaypointCommand waypoint_command);
 
-  robot_trajectory::RobotTrajectory jacobianPath3d(planning_scene_monitor::LockedPlanningSceneRW &planning_scene,
+  robot_trajectory::RobotTrajectory jacobianPath3d(planning_scene::PlanningScenePtr planning_scene,
```

Comment:
Why was it locked, and why is Locked no longer needed?

Reply:
the locked planning scene was from the PSM (emphasis on monitor) which mean it was a locked version of the globally shared PSM state. Now it's just a PS (no monitor) so it's not global state, so there's no concept of locking it or not. This removes lots of bug-prone spooky nonsense

Reply:
Excellent!